src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -3789,7 +3789,7 @@ void Unit::UpdateEnvironmentIfNeeded(const uint8 option)
             {
                 ground_z = (c->CanSwim() && liquidData.level > INVALID_HEIGHT) ? liquidData.level : temp;
                 bool canHover = c->CanHover();
-                isInAir = flyingBarelyInWater || (G3D::fuzzyGt(GetPositionZ(), ground_z + (canHover ? GetFloatValue(UNIT_FIELD_HOVERHEIGHT) : 0.0f) + GROUND_HEIGHT_TOLERANCE) || G3D::fuzzyLt(GetPositionZ(), ground_z - GROUND_HEIGHT_TOLERANCE)); // Can be underground too, prevent the falling
+                isInAir = flyingBarelyInWater || (G3D::fuzzyGt(GetPositionZ(), ground_z + (canHover ? GetFloatValue(UNIT_FIELD_HOVERHEIGHT) : 0.0f) + 0.75f) || G3D::fuzzyLt(GetPositionZ(), ground_z - 1.f)); // Can be underground too, prevent the falling
```

Comment:
@Footman  shall we wrap these magic numbers into constants? so they have a name that describes what they are

Reply:
Indeed, but I'm also curious why those specific values. What 0.75 and 1.f are exactly/practically?

Reply:
this part can be reverted. but i found often Z calculattion mistake in range of 1.f and not 0.5

Reply:
Could be it related to the dimension of the unit? maybe we could use a more dynamic value here such as the collisionheight

Reply:
> Could be it related to the dimension of the unit? maybe we could use a more dynamic value here such as the collisionheight

i dont know this. i just see pathgenerator sometimes placing points too high in the middle of the curve

Reply:
if you are not sure, please use a constant anyway, we worked a lot to remove all the hardcoded floats and it's not so good to add more of them now